toolchain/check/member_access.cpp

```diff
@@ -410,6 +409,8 @@ auto PerformCompoundMemberAccess(Context& context, Parse::NodeId node_id,
           member.type_id())) {
     member_id = PerformImplLookup(context, node_id, base_type_const_id,
                                   *assoc_type, member_id);
+  } else if (context.types().Is<SemIR::TupleType>(base_type_id)) {
```

Comment:
I'm not sure if this is the right way to do things.

Reply:
Yea, this fails. 

```
fn Main() {
  var non_tuple: [i32; 1] = (1,);
  var first: i32 = non_tuple.0;
}
```

Reply:
I understand the desire to use `types().Is` here, but I'm going to suggest making use of the already-fetched type constant:

```suggestion
  } else if (context.constant_values().GetAsInst(base_type_const_id).Is<SemIR::TupleType>()) {
```

Reply:
It seems like the method `GetAsInst` does not exist in `ConstantValueStore`. I can try implementing it, but I'm not sure how.

Reply:
@jonmeow sorry for the ping, i think you might have overlooked this